plugins/Referrers/Controller.php

```diff
@@ -87,19 +88,58 @@ public function getEvolutionGraph($typeReferrer = false, array $columns = array(
             if ($typeReferrer === false) {
                 $typeReferrer = Common::getRequestVar('typeReferrer', Common::REFERRER_TYPE_DIRECT_ENTRY);
             }
-            $label = self::getTranslatedReferrerTypeLabel($typeReferrer);
-            $total = $this->translator->translate('General_Total');
 
             if (!empty($view->config->rows_to_display)) {
                 $visibleRows = $view->config->rows_to_display;
             } else {
-                $visibleRows = array($label, $total);
+                $visibleRows = array($typeReferrer, 'total');
             }
 
-            $view->requestConfig->request_parameters_to_modify['rows'] = $label . ',' . $total;
+            $view->requestConfig->request_parameters_to_modify['rows'] = $typeReferrer . ',total';
         }
-        $view->config->row_picker_match_rows_by = 'label';
-        $view->config->rows_to_display = $visibleRows;
+
+        // We need to configure a custom set of selectable rows
+        // by default this would be generated using the label, but as the label is different in each language we use the
+        // referrer_type metadata for identification instead. This is the same across languages.
+        $view->config->filters[] = function ($dataTable) use ($view, $visibleRows) {
```

Comment:
This method is already really big imo, and now you have added a closure.

Would you consider extracting this filter to its own filter class so it's a little easier to read and easier to test?

Reply:
Generally I agree, but those config filters can either be provided as closure or as a class name. Providing a class name only wouldn't work in this case, as the filter is used to manipulate the view, which we can't pass through when using a class name only. Also this filter only works correctly with the view manipulation code above and below it, so moving it away might actually make it harder to understand it.

Reply:
We have it in multiple places that way. It's used when the view needs to be manipulated based on the values of the datatable. The problem is that you can't directly access the datatable at this stage, as it was not yet loaded. So this is solved by using a filter, which will be executed when loading the datatable. This might not be an ideal solution, but unfortunately currently the only we have, without doing a very big refactoring around how views and the datatables are handled.
But I already pushed an update, which will move that code to the view, where that workaround isn't needed, and it will be reusable.